### ISSUE TITLE
WriteGetObjectResponse - 405 MethodNotAllowed: The specified method is not allowed against this resource.

### ISSUE THREAD

**Issue by micheleriso** [aws/aws-sdk-js#3675]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js/issues) and didn't find any solution
  
**Describe the bug**
When using the WriteGetObjectResponse method, I get the following error `MethodNotAllowed: The specified method is not allowed against this resource.`

**Is the issue in the browser/Node.js?**
Node.js

**If on Node.js, are you running this on AWS Lambda?**
Yes

**Details of the browser/Node.js version**
Lambda Node Runtime 14.x

**SDK version number**
v2.870.0

**To Reproduce (observed behavior)**
```
const AWS = require('aws-sdk');
const s3 = new AWS.S3();

exports.handler = async (event, context) => {

  const objectContext = event['getObjectContext']
  const RequestRoute = objectContext['outputRoute']
  const RequestToken = objectContext['outputToken']

  try {
    const res = await s3.writeGetObjectResponse({
      RequestRoute,
      RequestToken,
      Body: "TEST",
      StatusCode: 200

    }).promise()

    console.log("Write Get Object SUCCESS", res)
  } catch (e) {
    console.error("WRITE GET OBJECT FAILED", e)
  }

  return {
    statusCode: 200
  }
};
```

**Expected behavior**
It should return `TEST` however it throws
```
2021-03-24T12:15:46.042Z	df108614-2c61-46a6-87a4-965e5fcbed95	ERROR	WRITE GET OBJECT FAILED MethodNotAllowed: The specified method is not allowed against this resource.
    at Request.extractError (/var/task/node_modules/aws-sdk/lib/services/s3.js:712:35)
    at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/var/task/node_modules/aws-sdk/lib/request.js:688:14)
    at Request.transition (/var/task/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/var/task/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /var/task/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:690:12)
    at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:116:18) {
  code: 'MethodNotAllowed',
  region: null,
  time: 2021-03-24T12:15:46.041Z,
  requestId: 'RQKY7099ES64PZRZ',
  extendedRequestId: '2LO2pWksxx8RNSsko7VL299wDuA65uGsOZH6SaHpCDvBWRDmw+x1khsJYZ2ZPGU5lXvY/R2Jt4g=',
  cfId: undefined,
  statusCode: 405,
  retryable: false,
  retryDelay: 3.0760893896501385
}

```

**Comment by micheleriso** [comment]
There is a workaround that can be used to temporary fix this issue

```
/** START Workaround for bug #3675 **/
const s3 = new AWS.S3({ region: "us-east-1", endpoint: "s3-object-lambda.us-east-1.amazonaws.com" });
s3.__proto__.api.endpointPrefix = "s3-object-lambda";
/** END Workaround for bug #3675 **/
```

I've documented the usage in my [medium blog](https://micheleriso.medium.com/serverless-framework-use-aws-s3-object-lambda-to-resize-images-on-the-fly-868a37017f3c) if that can help

**Comment by joe-keane** [comment]
That's for you @micheleriso, your sample was quite helpful.
Meanwhile, I notice this issue was fixed in 2.876.0.
I got a sample running and no workaround required.

**Comment by micheleriso** [comment]
Thanks Joe, it now works perfectly. I've updated my article to reflect that.
